src/foremast/pipeline/create_pipeline_kubernetes.py

```diff
@@ -0,0 +1,196 @@
+#   Foremast - Pipeline Tooling
+#
+#   Copyright 2018 Gogo, LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Create Pipelines for Spinnaker."""
+import json
+from pprint import pformat
+import requests
+from ..utils import check_managed_pipeline, get_all_pipelines, get_template
+from ..exceptions import SpinnakerPipelineCreationFailed
+from .clean_pipelines import delete_pipeline
+from ..consts import API_URL, GATE_CA_BUNDLE, GATE_CLIENT_CERT, DEFAULT_RUN_AS_USER
+from .create_pipeline import SpinnakerPipeline
+
+
+class SpinnakerPipelineKubernetesPipeline(SpinnakerPipeline):
+    """Manipulate Spinnaker Pipelines.
+
+    Args:
+        app (str): Application name.
+        trigger_job (str): Jenkins trigger job.
+        base (str): Base image name (i.e: fedora).
+        prop_path (str): Path to the raw.properties.json.
+    """
+
+    def render_wrapper_kubernetes(self, data):
+        """Generate the base Pipeline wrapper.
+
+        This renders the non-repeatable stages in a pipeline, like jenkins, baking, tagging and notifications.
+
+        Args:
+            data (dict): A dictionary with data to be used in the template
+
+        Returns:
+            dict: Rendered Pipeline wrapper.
+        """
+
+        # Contains generic wrapper (non-stage) pipeline configs
+        wrapper = get_template(template_file='pipeline/pipeline_wrapper.json.j2', data=data, formats=self.generated)
+        wrapper = json.loads(wrapper)
+        return wrapper
+
+    def create_pipeline(self):
+        """Main wrapper for pipeline creation.
+        1. Cleans up existing pipelines
+        2. Determines which environments and pipelines are needed
+        3. Created one pipeline PER environment given
+        4. Renders pipeline json from the template matching deployment_type given
+        5. Runs post_pipeline for each pipeline generated
+
+        Args:
+        Returns:
+            bool: True if pipelines were created successfully
+        """
+
+        pipeline_envs = self.environments
+        self.log.debug('Envs from pipeline.json: %s', pipeline_envs)
+
+        deleted_pipelines = self.clean_kubernetes_pipelines(pipeline_envs)
+        self.log.debug('Deleted %s managed pipelines: %s', len(deleted_pipelines), deleted_pipelines)
```

Comment:
This is likely better put in `clean_kubernetes_pipelines()`.
And if that's the case, there's no need to capture the output.